app/models/user.rb

```diff
@@ -6,6 +6,10 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
 
+  validates :password, confirmation: true
+  validates :password_confirmation, presence: true
+  validates :email, format: URI::MailTo::EMAIL_REGEXP
```

Comment:
As Devise already includes an email regex, it is recommended to reuse the same (to avoid some edge cases when 1 regex allow `x@y.z` when the other regex would require at least 2 chars before the `@` :harold:

You can get the Devise regex with `Devise.email_regexp`.
This regex is configured in `engines/ewa_user/config/initializers/devise.rb` with `config.email_regexp` ;-)